packages/react-instantsearch-core/src/widgets/Index.tsx

```diff
@@ -125,7 +125,9 @@ class Index extends Component<InnerProps, State> {
 
 const IndexWrapper: React.FC<Props> = props => (
   <InstantSearchConsumer>
-    {contextValue => <Index contextValue={contextValue} {...props} />}
+    {contextValue => (
+      <Index contextValue={contextValue} indexId={props.indexName} {...props} />
```

Comment:
This automatic overriding seems logical to me, but I was still confused seeing this written first, what about something like this:

```ts
const IndexWrapper: React.FC<Props> = props => {
  const inferredIndexId = props.indexName;
  return (
    <InstantSearchConsumer>
      {contextValue => (
        <Index contextValue={contextValue} indexId={inferredIndexId} {...props} />
      )}
    </InstantSearchConsumer>
  );
};
```

This makes it somewhat clearer to me it gets overriden